sdk/core/http/package.json

```diff
@@ -0,0 +1,184 @@
+{
+  "name": "@azure/core-http",
```

Comment:
To follow the repo structuring the folder should be named after the package. So in this case the it should be under sdk\core\core-http.

Reply:
Yep, I'll rename the folder from `http` to `core-http`.